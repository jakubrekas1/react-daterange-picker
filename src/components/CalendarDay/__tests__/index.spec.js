import React from 'react';
import renderer from 'react-test-renderer';
import { CalendarDay } from '../index';

describe('Calendar day component', () => {
  it('should renders correctly - day between selected dates', () => {
    const Component = (
      <CalendarDay
        currentMonth={0}
        date={new Date(2019, 0, 7)}
        dateEnd={new Date(2019, 0, 12)}
        dateStart={new Date(2019, 0, 1)}
        onDayClick={() => null}
      />
    );
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly - day from other month', () => {
    const Component = (
      <CalendarDay
        currentMonth={0}
        date={new Date(2019, 1, 1)}
        dateEnd={new Date(2019, 0, 12)}
        dateStart={new Date(2019, 0, 1)}
        onDayClick={() => null}
      />
    );
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly - is selected', () => {
    const Component = (
      <CalendarDay
        currentMonth={0}
        date={new Date(2019, 0, 12)}
        dateEnd={new Date(2019, 0, 12)}
        dateStart={new Date(2019, 0, 1)}
        onDayClick={() => null}
      />
    );
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
