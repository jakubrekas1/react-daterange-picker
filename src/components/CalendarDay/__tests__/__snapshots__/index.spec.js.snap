// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar day component should renders correctly - day between selected dates 1`] = `
<div
  className="calendar__day calendar__day--is-between"
  onClick={[Function]}
>
  7
</div>
`;

exports[`Calendar day component should renders correctly - day from other month 1`] = `
<div
  className="calendar__day calendar__day--is-other-month"
  onClick={[Function]}
>
  1
</div>
`;

exports[`Calendar day component should renders correctly - is selected 1`] = `
<div
  className="calendar__day calendar__day--is-selected"
  onClick={[Function]}
>
  12
</div>
`;
